projects/saturn/src/main/java/io/fairspace/saturn/webdav/BaseResource.java

```diff
@@ -143,12 +143,28 @@ private void validateTarget(io.milton.resource.CollectionResource parent, String
             setErrorMessage(message);
             throw new BadRequestException(message);
         }
-        var existing = parent.child(name);
-        if (existing != null) {
-            throw new ConflictException(existing);
+        if (parent != null && parent.child(name) != null) {
+            var message = "Resource with the name " + name + " already exist in the specified directory.";
+            setErrorMessage(message);
+            throw new ConflictException(message);
         }
     }
 
+    private void renameLinkedEntity(String name) {
+        subject.getPropertyResourceValue(FS.linkedEntity)
```

Comment:
same as previous question, can this be done safely based on name?

Reply:
this one, just looks up the "linked entity" property of a subject, doesn't use name for filtering